perfkitbenchmarker/benchmarks/mysql_service_benchmark.py

```diff
@@ -0,0 +1,838 @@
+
+    """
+    logging.info('Preparing MySQL Service benchmarks for RDS.')
+
+    # First is to create another subnet in the same VPC as the VM but in a
+    # different zone. RDS requires two subnets in two different zones to create
+    # a DB instance, EVEN IF you do not specify multi-AZ in your DB creation
+    # request.
+
+    # Get a list of zones and pick one that's different from the zone VM is in.
+    new_subnet_zone = ''
+    get_zones_cmd = util.AWS_PREFIX + ['ec2', 'describe-availability-zones']
+    stdout, _, _ = vm_util.IssueCommand(get_zones_cmd)
+    response = json.loads(stdout)
+    all_zones = response['AvailabilityZones']
+    for zone in all_zones:
+      if zone['ZoneName'] != vm.zone:
+        new_subnet_zone = zone['ZoneName']
+        break
+
+    if new_subnet_zone == '':
+      raise DBStatusQueryError('Cannot find a zone to create the required '
+                               'second subnet for the DB instance.')
+
+    # Now create a new subnet in the zone that's different from where the VM is
+    logging.info('Creating a second subnet in zone %s', new_subnet_zone)
+    new_subnet = aws_network.AwsSubnet(new_subnet_zone, vm.network.vpc.id,
+                                       '10.0.1.0/24')
+    new_subnet.Create()
+    logging.info('Successfully created a new subnet, subnet id is: %s',
+                 new_subnet.id)
+    # Remember this so we can cleanup properly.
+    vm.extra_subnet_for_db = new_subnet
+
+    # Now we can create a new DB subnet group that has two subnets in it.
+    db_subnet_group_name = 'pkb%s' % FLAGS.run_uri
+    create_db_subnet_group_cmd = util.AWS_PREFIX + [
+        'rds',
+        'create-db-subnet-group',
+        '--db-subnet-group-name', db_subnet_group_name,
+        '--db-subnet-group-description', 'pkb_subnet_group_for_db',
+        '--subnet-ids', vm.network.subnet.id, new_subnet.id]
+    stdout, stderr, _ = vm_util.IssueCommand(create_db_subnet_group_cmd)
+    logging.info('Created a DB subnet group, stdout is:\n%s\nstderr is:\n%s',
+                 stdout, stderr)
+    vm.db_subnet_group_name = db_subnet_group_name
+
+    # open up tcp port 3306 in the VPC's security group, we need that to connect
+    # to the DB.
+    open_port_cmd = util.AWS_PREFIX + [
+        'ec2',
+        'authorize-security-group-ingress',
+        '--group-id', vm.group_id,
+        '--source-group', vm.group_id,
+        '--protocol', 'tcp',
+        '--port', '3306']
+    stdout, stderr, _ = vm_util.IssueCommand(open_port_cmd)
+    logging.info('Granted tcp 3306 ingress, stdout is:\n%s\nstderr is:\n%s',
+                 stdout, stderr)
+
+    # Finally, it's time to create the DB instance!
+    vm.db_instance_id = 'pkb-DB-%s' % FLAGS.run_uri
+    db_class = \
+        RDS_CORE_TO_DB_CLASS_MAP['%s' % FLAGS.mysql_svc_db_instance_cores]
+    vm.db_instance_master_user = MYSQL_ROOT_USER
+    vm.db_instance_master_password = _GenerateRandomPassword()
+
+    create_db_cmd = util.AWS_PREFIX + [
```

Comment:
Per offline conversation, I've added a TODO here for such refactoring which will be done later with purely refactoring. (I wanted to contain the size of this CL)